common/persistence/cassandra/metadata_store.go

```diff
@@ -125,15 +125,22 @@ func (m *MetadataStore) CreateNamespace(
 	ctx context.Context,
 	request *p.InternalCreateNamespaceRequest,
 ) (*p.CreateNamespaceResponse, error) {
+
 	query := m.session.Query(templateCreateNamespaceQuery, request.ID, request.Name).WithContext(ctx)
-	applied, err := query.MapScanCAS(make(map[string]interface{}))
+	existingRow := make(map[string]interface{})
+	applied, err := query.MapScanCAS(existingRow)
 	if err != nil {
 		return nil, serviceerror.NewUnavailable(fmt.Sprintf("CreateNamespace operation failed. Inserting into namespaces table. Error: %v", err))
 	}
+
 	if !applied {
-		return nil, serviceerror.NewNamespaceAlreadyExists("CreateNamespace operation failed because of uuid collision.")
+		// if the id with the same name exists in `namespaces_by_id`, fall through and either add a row in `namespaces` table
+		// or fail if name exists in that table already. This is to make sure we do not end up with a row in `namespaces_by_id`
+		// table and no entry in `namespaces` table
+		if name, ok := existingRow["name"]; !ok || name != request.Name {
+			return nil, serviceerror.NewNamespaceAlreadyExists("CreateNamespace operation failed because of uuid collision.")
```

Comment:
I think we should add more detail here, something like: "CreateNamespace with name %q and id %q failed because another namespace with name %q already exists with the same id". Anything that includes the id and both names.